test/validation/api/ipsec/ipsec_test_out.c

```diff
@@ -1138,6 +1138,128 @@ static void test_out_ipv4_udp_esp_null_sha256(void)
 	ipsec_sa_destroy(sa);
 }
 
+static void test_sa_info(void)
+{
+	uint32_t src = IPV4ADDR(10, 0, 111, 2);
+	uint32_t dst = IPV4ADDR(10, 0, 222, 2);
+	odp_ipsec_tunnel_param_t tunnel_out;
+	odp_ipsec_tunnel_param_t tunnel_in;
+	odp_ipsec_sa_param_t param_out;
+	odp_ipsec_sa_param_t param_in;
+	odp_ipsec_capability_t capa;
+	odp_ipsec_sa_info_t sa_info;
+	odp_ipsec_sa_t sa_out;
+	odp_ipsec_sa_t sa_in;
+	uint64_t seq_num;
+
+	CU_ASSERT_EQUAL(0, odp_ipsec_capability(&capa));
+
+	memset(&tunnel_out, 0, sizeof(tunnel_out));
+	memset(&tunnel_in, 0, sizeof(tunnel_in));
+
+	tunnel_out.type = ODP_IPSEC_TUNNEL_IPV4;
+	tunnel_out.ipv4.src_addr = &src;
+	tunnel_out.ipv4.dst_addr = &dst;
+
+	ipsec_sa_param_fill(&param_out,
+			    false, false, 0x4a2cbfe3, &tunnel_out,
+			    ODP_CIPHER_ALG_AES_GCM, &key_mcgrew_gcm_3,
+			    ODP_AUTH_ALG_AES_GCM, NULL,
+			    &key_mcgrew_gcm_salt_3, NULL);
+
+	sa_out = odp_ipsec_sa_create(&param_out);
+
+	CU_ASSERT_NOT_EQUAL_FATAL(ODP_IPSEC_SA_INVALID, sa_out);
+
+	ipsec_sa_param_fill(&param_in,
+			    true, false, 0x4a2cbfe3, &tunnel_in,
+			    ODP_CIPHER_ALG_AES_GCM, &key_mcgrew_gcm_3,
+			    ODP_AUTH_ALG_AES_GCM, NULL,
+			    &key_mcgrew_gcm_salt_3, NULL);
+
+	sa_in = odp_ipsec_sa_create(&param_in);
+
+	memset(&sa_info, 0, sizeof(sa_info));
+	CU_ASSERT_EQUAL_FATAL(0, odp_ipsec_sa_info(sa_out, &sa_info));
+
+	CU_ASSERT_EQUAL(sa_info.param.dir, param_out.dir);
+	CU_ASSERT_EQUAL(sa_info.param.proto, param_out.proto);
+	CU_ASSERT_EQUAL(sa_info.param.mode, param_out.mode);
+
+	CU_ASSERT_EQUAL(sa_info.param.crypto.cipher_alg,
+			param_out.crypto.cipher_alg);
+	CU_ASSERT_EQUAL(sa_info.param.crypto.auth_alg,
+			param_out.crypto.auth_alg);
+	CU_ASSERT_EQUAL(sa_info.param.opt.udp_encap, param_out.opt.udp_encap);
+	CU_ASSERT_EQUAL(sa_info.param.spi, param_out.spi);
+	CU_ASSERT_EQUAL(sa_info.param.opt.esn, param_out.opt.esn);
+	CU_ASSERT_EQUAL(sa_info.param.opt.udp_encap, param_out.opt.udp_encap);
+	CU_ASSERT_EQUAL(sa_info.param.opt.copy_dscp, param_out.opt.copy_dscp);
+	CU_ASSERT_EQUAL(sa_info.param.opt.copy_flabel, param_out.opt.copy_flabel);
+	CU_ASSERT_EQUAL(sa_info.param.opt.copy_df, param_out.opt.copy_df);
+
+	if (param_out.mode == ODP_IPSEC_MODE_TUNNEL) {
```

Comment:
Tunnel mode SA was created, so this should be an assert.

Reply:
Ack.